### PR TITLE
Move iframe transport logic into transport.js

### PR DESCRIPTION
**Pull request by lannka**
#15729

### CODE REVIEW

File: extensions/amp-analytics/0.1/test/test-transport.js [hunk 1]
```diff
@@ -170,4 +170,51 @@ describes.realWin('amp-analytics.transport', {
       });
     });
   });
+
+  describe('iframe transport', () => {
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/test/test-transport.js]: done

File: extensions/amp-analytics/0.1/transport.js [hunk 2]
```diff
@@ -82,6 +97,45 @@ export class Transport {
...
+        'for amp-analytics tag with iframe transport.');
+
+    this.iframeTransport_ = new IframeTransport(
+        win, type, this.options_, ampAdResourceId);
+  }
+
+  /**
+   * Deletes iframe transport.
+   */
+  deleteIframeTransport() {
```
[the remark this answers is not in the document]
Comment (zhouyx) [reply extensions/amp-analytics/0.1/transport.js]: Well that's not something `amp-analytics.js` should care. I agree that we should be explicit within the Transport class, but not to whom simply want to dispose all Transport on unlayout.

LGTM but still not convinced : )

[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/transport.js]: The life cycle of the iframe is right now directly controlled by amp-analytics. We can only delegate it to transport.js if we also introduce those lifecycle callbacks in transport.js

[the remark this answers is not in the document]
Comment (zhouyx) [reply extensions/amp-analytics/0.1/transport.js]: However if we were to change `amp-analytics.js` why would I care about `iframeTransport`, thus `deleteIframeTransport`. All I need to understand is that there is a `Transport` instance that is created and it needs to be disposed to free resources at some life cycle event. And to that `dispose()` would be just fine.

[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/transport.js]: as I've pointed out, the debate is who should be responsible for managing the lifecycle of the iframe. At this point, transport itself does not really have a lifecycle.

**Pull request merged**